### PR TITLE
FourierMMD Refactor + Optimization

### PR DESCRIPTION
**Pull request by RobGeada**
This PR changes some of the data structures in the FourierMMD metric codebase to better align with Java principles, as well as changes some algorithm flows to better optimize computation.

### CODE REVIEW

File: explainability-service/src/main/java/org/kie/trustyai/service/payloads/metrics/drift/fouriermmd/FourierMMDMetricRequest.java [hunk 1]
```diff
@@ -26,8 +25,8 @@ public class FourierMMDMetricRequest extends DriftMetricRequest {
     private FourierMMDParameters parameters = new FourierMMDParameters();
     private double gamma = 2.0;
 
-    @JsonDeserialize(using = FourierMMDValuesDeserializer.class)
-    private Map<String, Object> fitting;
+    //@JsonDeserialize(using = FourierMMDValuesDeserializer.class)
```
Comment (danielezonca) [inline explainability-service/src/main/java/org/kie/trustyai/service/payloads/metrics/drift/fouriermmd/FourierMMDMetricRequest.java]: Let's remove the commented code

File: explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/drift/FourierMMDEndpoint.java [hunk 2]
```diff
@@ -33,6 +30,8 @@
         "distance between mean embeddings of features from the test dataframe and the training dataframe.")
 @Path("/metrics/drift/fouriermmd")
 public class FourierMMDEndpoint extends DriftEndpoint<FourierMMDMetricRequest> {
+    final Random rng = new Random();
```
Comment (danielezonca) [inline explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/drift/FourierMMDEndpoint.java]: I'm not a fan of having this logic directly in the endpoint code but even moving this inside `FourierMMD.precompute` is not ideal, overall we might consider to refactor with another ticket the whole concept with something like `PerturbationContext` to clearly wrap and control the random aspect.
Wdyt?

**Review by ruivieira — APPROVED** [review]

**Review by tteofili — APPROVED** [review]
LGTM

**Review by danielezonca — APPROVED** [review]

File: explainability-service/src/main/java/org/kie/trustyai/service/payloads/metrics/drift/fouriermmd/FourierMMDMetricRequest.java (hunk 1 above)
Comment (RobGeada) [reply explainability-service/src/main/java/org/kie/trustyai/service/payloads/metrics/drift/fouriermmd/FourierMMDMetricRequest.java]: Done

File: explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/drift/FourierMMDEndpoint.java (hunk 2 above)
Comment (RobGeada) [reply explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/drift/FourierMMDEndpoint.java]: replaced with fixed int

**Pull request merged**